### ISSUE TITLE
"transfer matrix has 0 elements"  in TBtrans

### ISSUE THREAD

**Issue by AsymmetryChou** [zerothi/sisl#643]
**Describe the issue**
Dear developers,
           Hello!  I encounter a problem when run TBtrans for calculating TB model of an carbon chain with 12 atoms, where 4 for central part and 4 for left/right electrode resepctively.  
-  The chain along Z-direction are placed with  0.16 nm between nearest atoms. 
-  I use sisl to finish the structure-read and electronic structure set up. (read from VASP structure file and set onsite/hopping in sisl) 
    As I'm considering a toy model, I take the nearest interation approximation on hopping element.
- After write nc files, I run TBtrans and have the problem:`Elec: transfer matrix has 0 elements. The self-energy cannot be calculated. Please check your electrode electronic structure.`
- I have checked the periodic condition for leads: In RUN.fdf, for left -A3 and for right +A3.
`
Hamiltonian{non-zero: 12, orthogonal: True,
 Spin{unpolarized, kind=f},
 Geometry{na: 4, no: 4,
  Atoms{species: 1,
   Atom{C, Z: 6, mass(au): 12.01070, maxR: -1.00000,
    Orbital{R: -1.00000, q0: 0.0}
   }: 4,
  },
  maxR: -1.00000,
  Lattice{nsc: [1 1 3],
   origin=0.0000, 0.0000, 0.0000,
   A=[10.0000, 0.0000, 0.0000],
   B=[0.0000, 10.0000, 0.0000],
   C=[0.0000, 0.0000, 6.4000],
   bc=[Periodic,
       Periodic,
       Periodic]
  }
 }
}
`

I have no idea where the problem is. The lead( with 4 atoms) Hamiltonian elements are:
`[[-0.50120997  0.02240228  0.          0.02240228]
 [ 0.02240228 -0.50120997  0.02240228  0.        ]
 [ 0.          0.02240228 -0.50120997  0.02240228]
 [ 0.02240228  0.          0.02240228 -0.50120997]]`
          
Thank you very much!

**Version details**
Run the below code and add to issue (if an issue is relevant for the issue):
3.9.18 (main, Sep 11 2023, 13:41:44) 
[GCC 11.2.0]
0.14.3.dev54+g83feb814c

**Comment by AsymmetryChou** [comment]
The RUN.fdf is :

TBT.HS chain.nc
TBT.k [50 50 1]
TBT.DOS.Gf true
TBT.DOS.A true
%block TBT.Elec.Left
  HS lead_L.nc
  semi-inf-direction -A3
  electrode-position 1
%endblock TBT.Elec.Left
%block TBT.Elec.Right
  HS lead_R.nc
  semi-inf-direction +A3
  electrode-position end -1
%endblock TBT.Elec.Right

**Comment by AsymmetryChou** [comment]
I think I have solved this problem. The mistake above may result from ignoring hopping elements between unit cells.
The interesting thing now is that:
the Lead Hamiltonina  is ( read by sisl.Hamiltonian.H[i,j] ,i/j range from 0 to 3)
`
[[-0.50120997  0.02240228  0.          0.        ]
 [ 0.02240228 -0.50120997  0.02240228  0.        ]
 [ 0.          0.02240228 -0.50120997  0.02240228]
 [ 0.          0.          0.02240228 -0.50120997]]
`
which contains 10 non-zero elements.  However, if I print out the Lead Hamiltonian.nnz,  it shows 12 non-zero element.  So it hides the 2 elements corresponding connection between nearest unit cells. That's the reason I encountered the above mistake.
